app/javascript/bundles/RestaurantsApp/components/RestaurantCard.jsx

```diff
@@ -10,38 +25,93 @@ export default class RestaurantCard extends React.Component {
      */
     constructor(props) {
         super(props);
+
+        this.state = {
+            rateModalIsOpen: false,
+            userRating: 0,
+        }
+
+        this.openRateModal = this.openRateModal.bind(this);
+        this.afterOpenModal = this.afterOpenModal.bind(this);
+        this.closeRateModal = this.closeRateModal.bind(this);
     }
 
     static propTypes = {
         restaurant: PropTypes.object.isRequired,
+        onRestaurantChange: PropTypes.func.isRequired,
     };
 
+    openRateModal() {
+        this.setState({rateModalIsOpen: true});
+    }
+
+    afterOpenModal() {
+        // references are now sync'd and can be accessed.
+        //this.subtitle.style.color = '#f00';
+    }
+
+    closeRateModal() {
+        this.setState({rateModalIsOpen: false});
+    }
+
 
     render() {
         const restaurant = this.props.restaurant;
         const cuisineTypes = restaurant.cuisine_types.map((cuisineType) => cuisineType.cuisine);
         return (
             <div key={restaurant.id} className="card restaurant-card">
                 <div className="card-block text-center">
-                    <img className="restaurant-logo" src={restaurant.logo}/>
-                    <h4 className="card-title">{restaurant.name}</h4>
-
-                    <div className="card-body">
-                        <div>{restaurant.address}</div>
-                        {restaurant.speed && (<div>{restaurant.speed} minute delivery</div>)}
-                        {restaurant.kosher === true && (<div>Kosher</div>)}
-                        {restaurant.rating && (
-                            <StarsRating
-                                rating={parseFloat(restaurant.rating)}
-                                interactiveRating={false}
-                                numStars={MAX_RESTAURANT_RATING}
-                            />
-                        )}
-                        {restaurant.accepts_10bis && (
-                            <img className="icon-10-bis restaurant-icon"/>
-                        )}
-                        <div>
-                            {cuisineTypes}
+                    <div className="image-section">
+                        <img className="restaurant-logo" src={restaurant.logo}/>
+                    </div>
+                    <div className="details-section">
+                        <h4 className="card-title">{restaurant.name}</h4>
+
+                        <div className="card-body">
+                            <div>{restaurant.address}</div>
+                            {restaurant.speed && (<div>{restaurant.speed} minute delivery</div>)}
+                            <div>
+                                {cuisineTypes}
+                            </div>
+                            <div className="rest-sub-section">
+                                <div className="rest-sub-item">
+                                    {restaurant.kosher === true && (
```

Comment:
Is it necessary to test that it's `true`? You could probably omit the `=== true` part.